rosco-manifests/src/test/groovy/com/netflix/spinnaker/rosco/manifests/kustomize/KustomizeTemplateUtilsSpec.groovy

```diff
@@ -0,0 +1,195 @@
+                referenceBase + "examples/ldap/base/deployment.yaml",
+                referenceBase + "examples/ldap/base/kustomization.yaml",
+                referenceBase + "examples/ldap/base/env.startup.txt"
+        ].sort()
+    }
+
+    def "getFilesFromSameFolder returns a list of files where one of the resources is referencing to a kustomization"() {
+        given:
+        def referenceBase = "https://api.github.com/repos/kubernetes-sigs/kustomize/contents/"
+        def clouddriverService = Mock(ClouddriverService)
+        def kustomizationFileReader = Mock(KustomizationFileReader)
+        def kustomizationTemplateUtils = new KustomizeTemplateUtils(kustomizationFileReader, clouddriverService)
+        def baseArtifact = Artifact.builder()
+                .name("examples/helloWorld/kustomization.yaml")
+                .reference(referenceBase + "examples/helloWorld/kustomization.yaml")
+                .artifactAccount("github")
+                .build()
+
+        when:
+        def filesToFetch = kustomizationTemplateUtils.getFilesFromArtifact(baseArtifact)
+
+        then:
+        // the base artifact supplies deployment.yml, service.yml and configMap.yaml
+        1 * kustomizationFileReader.getKustomization(_ as Artifact, _ as String) >> { Artifact a, String s ->
+            Kustomization k = new Kustomization()
+            if (a.getName() == "examples/helloWorld") {
+                k.setResources(Arrays.asList("deployment.yaml", "service.yaml", "configMap.yaml"))
+            }
+            k.setKustomizationFilename(s)
+            return k
+        }
+        filesToFetch.sort() == [
+                referenceBase + "examples/helloWorld/deployment.yaml",
+                referenceBase + "examples/helloWorld/service.yaml",
+                referenceBase + "examples/helloWorld/configMap.yaml",
+                referenceBase + "examples/helloWorld/kustomization.yaml"
+        ].sort()
+    }
+
+    def "getFilesFromMixedFolders returns a list of files where one of the resources is referencing another kustomization (5)"() {
+        given:
+        def referenceBase = "https://api.github.com/repos/kubernetes-sigs/kustomize/contents/"
+        def clouddriverService = Mock(ClouddriverService)
+        def kustomizationFileReader = Mock(KustomizationFileReader)
+        def kustomizationTemplateUtils = new KustomizeTemplateUtils(kustomizationFileReader, clouddriverService)
+        def baseArtifact = Artifact.builder()
+                .name("examples/multibases/kustomization.yaml")
+                .reference(referenceBase + "examples/multibases/kustomization.yaml")
+                .artifactAccount("github")
+                .build()
+
+        when:
+        def filesToFetch = kustomizationTemplateUtils.getFilesFromArtifact(baseArtifact)
+
+        then:
+        7 * kustomizationFileReader.getKustomization(_ as Artifact, _ as String) >> { Artifact a, String s ->
```

Comment:
I'd suggest dropping the precise number of times this is called from the test, as this often leads to overspecified tests.  For mutating operations it's often important that we know exactly how many times they are called---but for more of a 'getter' like this it's probably better to just use a stub (that returns test data but doesn't care about number of calls) instead of a mock that enforces the exact number of calls.